### DISCUSSION TITLE
How to retry at a fixed interval?

### DISCUSSION THREAD

**Discussion by fighterhit** [go-resty/resty#832] [Q&A] [answered]
How to retry 3 times with a fixed interval of 300ms? I found that when using the following code, the retry interval is 3s and the number of retries is 4.
```
        httpClient = resty.New().SetRetryCount(3).
                        SetRetryWaitTime(time.Millisecond * 200).
                        SetRetryMaxWaitTime(time.Millisecond * 200).
                        AddRetryCondition(func(response *resty.Response, err error) bool {
                        return response.IsError() || err != nil
                })

	resp, err := httpClient.R().SetHeaders(ws.headerMap).SetBody(body).SetResult(&ApiResult{}).Post(endpoint)
...

```

```
...
2024/08/28 22:10:03.012050 ERROR RESTY Post "http://xxx.yyy/api/v1/training/rl/alert": EOF, Attempt 1
2024/08/28 22:10:06.220787 ERROR RESTY Post "http://xxx.yyy/api/v1/training/rl/alert": EOF, Attempt 2
2024/08/28 22:10:09.430459 ERROR RESTY Post "http://xxx.yyy/api/v1/training/rl/alert": EOF, Attempt 3
2024/08/28 22:10:12.638736 ERROR RESTY Post "http://xxx.yyy/api/v1/training/rl/alert": EOF, Attempt 4
...
```

**Comment by jeevatkm** [comment, marked as answer]
@fighterhit Currently, Resty does `Backoff` retries, not a fixed interval. In the upcoming Resty v3, I aim to bring a pluggable retry algorithm.

**Reply by fighterhit** [reply]
@jeevatkm  Thanks for the prompt reply. So now we can't actually control the retry interval, right? Another thing that confuses me is what is the meaning and difference between `SetRetryWaitTime` and `SetRetryMaxWaitTime`? How does it affect the retry?

**Reply by jeevatkm** [reply]
As I mentioned before. Resty uses a backoff mechanism for retry. The wait time (min) and max wait time are used to calculate the interval for the delay.

Please refer to unexported methods `sleepDuration` and `jitterBackoff` on the file `retry.go`.

**Comment by jeevatkm** [comment]
In Resty v3 - https://resty.dev/docs/resilience/retry/#constant-delay
